components/sceneutil/riggeometry.cpp

```diff
@@ -160,9 +161,16 @@ bool RigGeometry::initFromParentSkeleton(osg::NodeVisitor* nv)
         }
     }
 
+    Bone2VertexMap bone2VertexMap;
     for (Vertex2BoneMap::iterator it = vertex2BoneMap.begin(); it != vertex2BoneMap.end(); ++it)
     {
-        mBone2VertexMap[it->second].push_back(it->first);
+        bone2VertexMap[it->second].push_back(it->first);
+    }
+
+    mBone2VertexVector.clear();
+    for (auto &pair : bone2VertexMap)
+    {
+        mBone2VertexVector.push_back(pair);
     }
```

Comment:
I still think using assign is better than using clear() and a loop.

Reply:
Done.